plugins/aggregators/histogram/README.md

```diff
@@ -0,0 +1,112 @@
+Internally buckets are presented as ranges with borders 0..bucketBorder: 0..1, 0..10, 0..50, …, 0..+Inf.
+So the value "+1" will be put into those buckets, in which the metric value fell with such ranges of buckets.
+
+Also, the algorithm of hit counting to buckets was implemented on the base of the algorithm, which is implemented in
+the Prometheus [client](https://github.com/prometheus/client_golang/blob/master/prometheus/histogram.go).
+
+### Configuration
+
+```toml
+# Configuration for aggregate histogram metrics
+[[aggregators.histogram]]
+  ## General Aggregator Arguments:
+  ## The period on which to flush & clear the aggregator.
+  period = "30s"
+  ## If true, the original metric will be dropped by the
+  ## aggregator and will not get sent to the output plugins.
+  drop_original = false
+
+  ## The example of config to aggregate histogram for all fields of specified metric.
+  [[aggregators.histogram.config]]
+  ## The set of buckets.
+  buckets = [0.0, 15.6, 34.5, 49.1, 71.5, 80.5, 94.5, 100.0]
+  ## The name of metric.
+  metric_name = "cpu"
+
+  ## The example of config to aggregate histogram for concrete fields of specified metric.
+  [[aggregators.histogram.config]]
+  ## The set of buckets.
+  buckets = [0.0, 10.0, 20.0, 30.0, 40.0, 50.0, 60.0, 70.0, 80.0, 90.0, 100.0]
+  ## The name of metric.
+  metric_name = "diskio"
+  ## The concrete fields of metric.
+  metric_fields = ["io_time", "read_time", "write_time"]
+```
+
+#### Explanation
+
+The field `metric_fields` is the list of metric parameters. For example, the metric `cpu` has the following parameters:
+usage_user, usage_system, usage_idle, usage_nice, usage_iowait, usage_irq, usage_softirq, usage_steal, usage_guest,
+usage_guest_nice.
+
+Note that histogram metrics will be pushed every `period` seconds. 
+As you know telegraf calls aggregator `Reset()` func each `period` seconds. Histogram aggregator ignores `Reset()` and continues to count hits. 
+
+#### Use cases
+
+You can specify parameters using two cases:
+
+ 1. The specifying only metric name. In this case all parameters of metric will be aggregated.
+ 2. The specifying metric name and concrete parameters.
+ 
+#### Some rules
+ 
+ - The setting of each histogram must be in separate section with title `aggregators.histogram.config`.
+
+ - The each value of bucket must be float value.
+ 
+ - Don\`t include the border bucket `+Inf`. It will be done automatically.
+ 
+### Measurements & Fields:
+
+The postfix `bucket` will be added to each parameter.
+
+- measurement1
+    - field1_bucket
+    - field2_bucket
+
+### Tags:
+
+All measurements have tag `le`. This tag has the border value of bucket. It means that the metric value is less or equal
+to the value of this tag. For example, let assume that we have the metric value 10 and the following buckets:
+[5, 10, 30, 70, 100]. Then the tag `le` will have the value 10, because the metrics value is passed into bucket with
+right border value `10`.
+
```

Comment:
You want so that we add some InfluxQL queries, right? Do you suggest doing this for a better understating? If so, could you provide us any plugin or aggregator, where we can take an example of queries?